fastserde/avro-fastserde-tests-common/src/test/java/com/linkedin/avro/fastserde/FastGenericSerializerGeneratorTest.java

```diff
@@ -562,6 +563,81 @@ public double getPrimitive(int index) {
     Assert.assertTrue(primitiveApiCalled.get());
   }
 
+  @Test(groups = {"serializationTest"})
+  public void shouldPassThroughByteBufferForArrayOfFloats() {
+    String arrayOfFloatFieldName = "array_of_float";
+    Schema recordSchema = createRecord("TestArrayOfFloats", new Schema.Field(arrayOfFloatFieldName, Schema.createArray(Schema.create(Schema.Type.FLOAT)), null, null));
+    GenericRecord record = new GenericData.Record(recordSchema);
+    record.put(arrayOfFloatFieldName, Arrays.asList(1.0f, 2.0f, 3.0f));
+
+    /**
+     * Deserialize it first by fast deserializer to check whether {@link BufferBackedPrimitiveFloatList} is being used or not.
+      */
+    GenericRecord decodedRecord = decodeRecordFast(recordSchema, dataAsBinaryDecoder(record));
+    Assert.assertTrue(decodedRecord.get(arrayOfFloatFieldName) instanceof BufferBackedPrimitiveFloatList);
+
+    class TestBufferBackedPrimitiveFloatList extends BufferBackedPrimitiveFloatList {
+
+      boolean writeFloatsCalled = false;
+      boolean writeFloatsByBackedBytesCalled = false;
+      public TestBufferBackedPrimitiveFloatList(BufferBackedPrimitiveFloatList floatList) {
+        super(0);
+        floatList.copyInternalState(this);
+      }
+
+      public void reset() {
```

Comment:
Can we call this `resetFlags` or something like that? I was a bit confused because I was thinking about the `GenericArray`'s `reset` API which is also involved in this commit, but unrelated to the code here...